src/lib/comms/sol-network-impl-zephyr.c

```diff
@@ -191,3 +191,33 @@ sol_network_link_get_name(const struct sol_network_link *link)
 {
     return NULL;
 }
+
+SOL_API int
+sol_network_hostname_pending_cancel(
+    struct sol_network_hostname_pending *handle)
+{
+    return -ENOTSUP;
+}
+
+SOL_API int
+sol_network_link_up(uint16_t link_index)
+{
+    SOL_WRN("Not implemented");
+    return -ENOSYS;
+}
+
+SOL_API int
+sol_network_link_down(uint16_t link_index)
+{
+    SOL_WRN("Not implemented");
+    return -ENOSYS;
+}
+
+SOL_API struct sol_network_hostname_pending *
+sol_network_get_hostname_address_info(const struct sol_str_slice hostname,
+    enum sol_network_family family, void (*host_info_cb)(void *data,
+    const struct sol_str_slice host, const struct sol_vector *addrs_list),
+    const void *data)
+{
+    return NULL;
```

Comment:
We could do it, at least a INF.
I was following the RIOT implementation and it's used by some samples. So we're kind of expecting it to fail, so not so sure if it should warn.